### ISSUE TITLE
extract regexes in tailwind.config.js don't seem to match?

### ISSUE THREAD

**Issue by erlandsona** [houseabsolute/tailwindcss-to-rust#3]
I pulled in tailwindcss-to-rust (not the macros) to use this for compile checked tailwind references in an actix-web serving raw html project. 

But the tailwind.config.js isn't picking up the Rust const references...

Here's the error
```
╰  npx tailwindcss --input src/assets/tailwind.css --output public/assets/tailwind_compiled.css

Rebuilding...

warn - No utility classes were detected in your source files. If this is unexpected, double-check the `content` option in your Tailwind CSS configuration.
warn - https://tailwindcss.com/docs/content-configuration
```
Here's the server.
```rs
#[actix_web::main]
async fn main() -> std::io::Result<()> {
    HttpServer::new(|| {
        App::new()
            .service(hello)
            .service(
                fs::Files::new("/", "./public")
                    .show_files_listing()
                    .use_last_modified(true),
            )
    })
    .bind(("0.0.0.0", 7373))?
    .run()
    .await
}


#[get("/")]
async fn hello() -> impl Responder {
    HttpResponse::Ok().body(html!(
       <html>
         <head>
            <title>"Stuff a ndthigns"</title>
            <link rel="stylesheet" href="/assets/tailwind_compiled.css" />
        </head>
         <body>
            <div class={C.bg.bg_rose_500}>"Is this "<i>{hey("italicised?")}</i></div>
         </body>
       </html>
    ))
}
```

**Comment by autarch** [comment]
I'm not sure why this is happening. That said, I haven't tested this code with the latest tailwindcss. Can you share your `tailwind.config.js` file? Or better yet, share the whole project so I can test the whole thing?

**Comment by autarch** [comment]
I'd also need to see the file you pass to `tailwindcss --input` if you can't share the whole project.

**Comment by autarch** [comment]
Ah, I figured it out. The regexes in the `tailwind.config.rs` were matching your trailing `}` character. Here's an updated version (that I'll add to the docs in the next release):

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: {
    files: ["index.html", "**/*.rs"],
    // You do need to copy this big block of code in, unfortunately.
    extract: {
      rs: (content) => {
        const rs_to_tw = (rs) => {
          if (rs.startsWith("two_")) {
            rs = rs.replace("two_", "2");
          }
          return rs
            .replaceAll("_of_", "/")
            .replaceAll("_p_", ".")
            .replaceAll("_", "-");
        };

        let classes = [];
        let class_re = /C\.[^ ]+\.([^\. {}]+)\b/g;
        let mod_re = /(?:M\.([^\. ]+)\s*,\s*)+C\.[^ ]+\.([^\. {}]+)\b/g;
        let matches = [...content.matchAll(mod_re)];
        if (matches.length > 0) {
          classes.push(
            ...matches.map((m) => {
              let pieces = m.slice(1, m.length);
              return pieces.map((p) => rs_to_tw(p)).join(":");
            })
          );
        }
        classes.push(
          ...[...content.matchAll(class_re)].map((m) => {
            return rs_to_tw(m[1]);
          })
        );
          console.log(classes);
        return classes;
      },
    },
  },
  theme: {
    extend: {},
  },
  plugins: [],
};
```

**Comment by autarch** [comment]
Actually, I think the regexes can be tightened even further. I'll do that for the next release, but what I shared above should work you for now.

**Comment by autarch** [comment]
Actually, it's even more complex now that I think about it. I think these regexes need to be customized for any templating system you use, if you're _not_ using the macros. I'm not sure what you're using to get that `html!` macro, but it looks like you interpolate with `{ ... }`, so you'd need regexes that match that.

**Comment by erlandsona** [comment]
I'm using the _super_ basic html-to-string-macro that's backed by syn-rsx but before I was using render.rs for templating...

Just getting back to this, and not sure if you just added it or if I just totally didn't read the docs but just noticed this bit
![image](https://user-images.githubusercontent.com/10421138/216846880-73cfa7dc-80d8-49aa-8c02-5958a74406b6.png)


And that totally makes sense. I'll give that a shot and see how I fair.

**Comment by autarch** [comment]
Yes, I just added that bit for the 0.2.0 release. I realized the docs described the case of using the macros as if that was the _only_ way to do it, which it's not. You should be able to adjust the regex above to match your case. I think all you have to do is replace the double curly braces with just one and it should work.

**Comment by erlandsona** [comment]
This is great!

Was worried coming from Elm & elm-ui I'd be missing out on the "inline styles" plus language native feel of doing elm-ui but with tailwindcss-to-rust it's a nice approximation to elm-ui.

Thanks a ton!

**Comment by autarch** [comment]
I'm glad it's working for you!

**Comment by erlandsona** [comment]
Ok, "working" is relative to the Stack Overflow I was getting today :grimacing:

Commenting out my use of the `C![]` macros in my templates got rid of it, so now I'm curious what's happenin' again, Sorry :sweat_smile:

**Comment by autarch** [comment]
This is mentioned in the README. There's some sort of weird interaction with Rust's compiler here. Try building with `--release` to see if that fixes it.
